### PR TITLE
Add crypto randomBytes and store logic

### PR DESCRIPTION
**Pull request by vderunov**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,20 +1,72 @@
...
+    throw error;
+  }
+};
 
-  console.log(walletAddress);
-  res.status(200).send({ message: 'Signup successful' });
+app.post('/signup', async (req, res) => {
+  if (!validateWalletAddress(req, res)) return;
+
+  const filePath = path.join(DATA_DIR, req.body.walletAddress);
```
Comment (noisekit) [inline server.js:56]:
```suggestion
  const filePath = path.join(DATA_DIR, req.body.walletAddress.toLowerCase());
```

File: server.js [hunk 2]
```diff
@@ -1,20 +1,72 @@
...
+  try {
+    if (await walletAddressStored(filePath)) {
+      res.status(200).send({ signature: await fs.readFile(filePath, 'utf8') });
+      return;
+    }
+
+    const randomBytes = await generateRandomHexString();
+    await storeWalletAddress(req.body.walletAddress, randomBytes);
```
Comment (noisekit) [inline server.js:64]:
```suggestion
    await storeWalletAddress(req.body.walletAddress.toLowerCase(), randomBytes);
```

**Review by noisekit — APPROVED** [review]
With a few small updates 👌

File: server.js (hunk 1 above)
Comment (vderunov) [reply server.js]: fixed

File: server.js [hunk 3]
```diff
@@ -1,20 +1,72 @@
...
+    return true;
+  } catch {
+    return false;
   }
+};
 
-  // Additional logic will be involved here
+const generateRandomHexString = async () => {
+  return new Promise((resolve) => {
+    crypto.randomBytes(256, (err, buf) => {
```
[the remark this answers is not in the document]
Comment (vderunov) [reply server.js]: fixed

**Pull request merged**